subcommands/version/updates.go

```diff
@@ -163,6 +163,7 @@ func (u FioctlUpdate) Do() error {
 	_ = f.Close()
 	sha := pb.sha.Sum(nil)
 	if !hmac.Equal(sha, u.Sha512) {
+		_ = os.Remove(tmpExe)
```

Comment:
So, the `tmpExe` is removed only if it is fully downloaded and its hash doesn't match expected one?
I thought about removing it if some other error happens during download, e.g. size of data being downloaded exceeded the expected one (the limit reader yields an error) or the `fioctl command` is interrupted by system signal (ctrl-c).
I think it's not critical, anyway the next attempt will rewrite it.

Reply:
the "limit reader" is put into the "tee reader". if that reader fails, we remove the file. In the event of an "under read", the sha won't match and that will remove the file.

ctrl-c is a problem but I think you are right - lets wait till we redo things for that